packages/cli/src/serve/output/render.rs

```diff
@@ -0,0 +1,277 @@
+
+        // Build the status bar.
+        let status_bar = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints([Constraint::Fill(1), Constraint::Fill(1)])
+            .split(body[2]);
+
+        // Specify borders
+        let border_sep = body[1];
+
+        Self {
+            _body: body,
+            console,
+            border_sep,
+            status_bar,
+        }
+    }
+
+    /// Render all decorations.
+    pub fn render_decor(&self, frame: &mut Frame) {
+        frame.render_widget(
+            Block::new()
+                .borders(Borders::TOP)
+                .border_style(Style::new().gray()),
+            self.border_sep,
+        );
+    }
+
+    /// Render the console and it's logs.
+    pub fn render_console(
+        &self,
+        frame: &mut Frame,
+        scroll: ScrollPosition,
+        messages: &[Message],
+    ) -> NumLinesWrapping {
+        // TODO: Fancy filtering support "show me only app logs from web"
+        let console = self.console[0];
+        let mut out_text = Text::default();
+
+        // Filter logs for current tab.
+        // Display in order they were created.
+        let msgs = messages.iter();
+
+        for msg in msgs {
+            for line in msg.content.lines() {
+                let text = line.into_text().unwrap_or_default();
+                for line in text.lines {
+                    // Don't add any formatting for cargo messages.
+                    let out_line = if msg.source != MessageSource::Cargo {
+                        let source = format!("[{}]", msg.source);
+                        let source_span = Span::from(source);
+                        let source_span = match msg.source {
+                            MessageSource::App(_) => source_span.light_cyan(),
+                            MessageSource::Dev => source_span.dark_gray(),
+                            MessageSource::Build => source_span.light_yellow(),
+                            MessageSource::Unknown => source_span.black(),
+                            MessageSource::Cargo => {
+                                unimplemented!("this shouldn't be reached")
+                            }
+                        };
+
+                        let level = format!(" {}: ", msg.level);
+                        let level_span = Span::from(level);
+                        let level_span = match msg.level {
+                            Level::TRACE => level_span.black(),
+                            Level::DEBUG => level_span.light_magenta(),
+                            Level::INFO => level_span.light_blue(),
+                            Level::WARN => level_span.light_yellow(),
+                            Level::ERROR => level_span.light_red(),
+                        };
+
+                        let mut out_line = vec![source_span, level_span];
+                        for span in line.spans {
+                            out_line.push(span);
```

Comment:
Per UX-review, we should only mark the first line with the source tag - all other lines should be at the right separation distance (as per measured by the first line) but not have the `[build]` tag.

```
[build] INFO: Optimized static asset /Users/jonkelley/Developm... [minified]                                                                              ▐
[build] INFO: Copying public assets to the output directory...                                                                                            ▐
[build] INFO: Running wasm-bindgen                                                                                                                        ▐
[build] WARN: The `web.resource` config has been deprecated in favor of head components and will be removed in a future release. Instead of including     ▐
assets in the config, you can include assets with the `asset!` macro and add them to the head with `head::Link` and `Script` components.                  ▐
[build] WARN: To migrate to head components, remove `web.resource.style` and include the following rsx in your root component:                            ▐
[build] WARN: ```rust                                                                                                                                     ▐
[build] WARN: rsx! {                                                                                                                                      ▐
[build] WARN:     head::Link { rel: "stylesheet", href: asset!(css("./assets/main.css")) }                                                                ▐
[build] WARN: }                                                                                                                                           ▐
```
[build] INFO: Optimized static asset /Users/jonkelley/Developm... [minified]                                                                              ▐
[build] INFO: Copying public assets to the output directory...                                                                                            ▐
[build] INFO: Running wasm-bindgen                                                                                                                        ▐
[build] WARN: The `web.resource` config has been deprecated in favor of head components and will be removed in a future release. Instead of including     ▐
assets in the config, you can include assets with the `asset!` macro and add them to the head with `head::Link` and `Script` components.                  ▐
[build] WARN: To migrate to head components, remove `web.resource.style` and include the following rsx in your root component:                            ▐
[build] WARN: ```rust                                                                                                                                     ▐
[build] WARN: rsx! {                                                                                                                                      ▐
[build] WARN:     head::Link { rel: "stylesheet", href: asset!(css("./assets/main.css")) }                                                                ▐
[build] WARN: }                                                                                                                                           ▐
```

To something like:

```
[build] WARN: The `web.resource` config has been deprecated in favor of head components and will be removed in a future release.
              To migrate to head components, remove `web.resource.style` and include the following rsx in your root component:                            
              ```rust                                                                                                                                     
              rsx! {                                                                                                                                      
                  head::Link { rel: "stylesheet", href: asset!(css("./assets/main.css")) }                                                                
              }                                                                                                                                           
```